### PR TITLE
Create exercise sum-of-multiples

### PR DESCRIPTION
**Pull request by PaulT89**
## Summary

Generated practice exercise Sum of Multiples.

Difficulty = 3.  Could be a 2 though.

## Checklist
- [x] If docs where changed run `./bin/configlet generate` to ensure all documents are properly generated.
- [x] CI is green

### CODE REVIEW

File: exercises/practice/sum-of-multiples/.meta/example.lisp [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+(in-package :sum-of-multiples)
+
+(defun sum (factors limit)
+ (let ((cleaned-factors (remove-if-not #'plusp factors)))
+   (apply #'+ (remove-duplicates (multiples-of cleaned-factors limit)))))
+
+(defun multiples-of (factors limit)
+  (when factors
+   (let ((factor (first factors))
```
Comment (verdammelt) [inline exercises/practice/sum-of-multiples/.meta/example.lisp:13]: Should these be put into the `loop`, perhaps with something like `with (factor . remaining) = factors`?

File: exercises/practice/sum-of-multiples/.meta/example.lisp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+         (remaining (rest factors)))
+     (loop for x from factor below limit by factor
+       collect x into output
+       finally (return (append output (multiples-of remaining limit)))))))
```
Comment (verdammelt) [inline exercises/practice/sum-of-multiples/.meta/example.lisp:17]: Why recursion? Why not iterate across the list of factors?

File: exercises/practice/sum-of-multiples/sum-of-multiples-test.lisp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+;; Enter the testing package
+(in-package :sum-of-multiples-test)
+
+;; Define and enter a new FiveAM test-suite
+(def-suite* sum-of-multiples-suite)
+
+(test no-multiples-within-limit
+    (let ((factors (list 3 5))
+          (limit 1))
+      (is (equal 0 (sum-of-multiples:sum factors limit)))))
```
Comment (verdammelt) [inline exercises/practice/sum-of-multiples/sum-of-multiples-test.lisp:21]: I think we should use `=` in these tests.

**Review by verdammelt — COMMENTED** [review]
Just some comments. Wondering if we should create some sort of "iteration" concept and put this in it.

File: exercises/practice/sum-of-multiples/sum-of-multiples-test.lisp (hunk 3 above)
Comment (PaulT89) [reply exercises/practice/sum-of-multiples/sum-of-multiples-test.lisp:21]: Agreed.

File: exercises/practice/sum-of-multiples/.meta/example.lisp (hunk 2 above)
Comment (PaulT89) [reply exercises/practice/sum-of-multiples/.meta/example.lisp:17]: Mostly just style choice.  I can rewrite it as a nested loop.

**Comment by PaulT89** [comment]
> Just some comments. Wondering if we should create some sort of "iteration" concept and put this in it.

Isn't iteration slightly different from loops?  I mean, the answer is 'yes' either way, but I'm just curious.

**Comment by verdammelt** [comment]
Iteration (to me) is a concept about how to well.. iterate - maybe through a sequence or via a counter (good old for-loop). Iteration can be implemented in multiple ways - Common Lisp has several built in iteration constructs but are generally `do` (and its kin) and `loop` and I guess recursion.

So maybe we need a concept of iteration with branches for `do` and `loop` (where each of those have sub-branches based upon functionality (i.e. `do times` vs `dolist` vs all the crazy `loop` `for` keywords...)

**Comment by PaulT89** [comment]
Okay.  I remember why I thought the term 'iterate' meant something else - I kept seeing it in [the loops section of the Common Lisp Cookbook](https://lispcookbook.github.io/cl-cookbook/iteration.html) and didn't realize that it's just another loop macro, just not a built-in one.

**Pull request merged**